openmdao/core/system.py

```diff
@@ -2967,7 +2972,19 @@ def list_inputs(self,
         meta = self._var_abs2meta
         inputs = []
 
+        if isinstance(tags, str):
+            tags = [tags, ]
+
         for var_name, val in iteritems(self._inputs._views):  # This is only over the locals
+
+            # Filter based on tags
+            if tags:
```

Comment:
Might consider making this filtering based on tags code into a function since it's replicated multiple times.